server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/CounterAddHandler.java

```diff
@@ -0,0 +1,164 @@
+package org.jboss.as.clustering.infinispan.subsystem;
+
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.OP_ADDR;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+import org.infinispan.counter.api.CounterConfiguration;
+import org.infinispan.counter.api.CounterConfiguration.Builder;
+import org.infinispan.counter.api.CounterType;
+import org.infinispan.counter.api.Storage;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.server.infinispan.spi.service.CacheContainerServiceName;
+import org.infinispan.server.infinispan.spi.service.CounterServiceName;
+import org.jboss.as.controller.AbstractAddStepHandler;
+import org.jboss.as.controller.AttributeDefinition;
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.dmr.ModelNode;
+import org.jboss.msc.service.Service;
+import org.jboss.msc.service.ServiceBuilder;
+import org.jboss.msc.service.ServiceController;
+import org.jboss.msc.service.ServiceTarget;
+import org.jboss.msc.value.InjectedValue;
+
+/**
+ * Add operation handler for /subsystem=infinispan/cache-container=clustered/counter=*
+ *
+ * @author Vladimir Blagojevic
+ *
+ */
+public class CounterAddHandler extends AbstractAddStepHandler implements  RestartableServiceHandler{
+
+    CounterAddHandler() {
+    }
+
+    @Override
+    protected void performRuntime(OperationContext context, ModelNode operation, ModelNode model)
+            throws OperationFailedException {
+
+        super.performRuntime(context, operation, model);
+
+        this.installRuntimeServices(context, operation, model, null);
+    }
+
+    @Override
+    public Collection<ServiceController<?>> installRuntimeServices(OperationContext context, ModelNode operation,
+            ModelNode containerModel, ModelNode cacheModel) throws OperationFailedException {
+
+        String counterName = getCounterName(operation);
+        String containerName = getContainerName(operation);
+        String counterType = getCounterType(operation);
+
+        Builder b = getBuilder(containerModel, counterType);
+        processModelNode(context, containerModel, b);
+        String name = CounterResource.COUNTER_NAME.resolveModelAttribute(context, containerModel).asString();
+        if (!counterName.equals(name)) {
+            throw new OperationFailedException("Counter node name and node's name attribute should be the same");
```

Comment:
@pruivo @remerson I added the check to make sure the name node and the attribute name are equal. No more inconsistent counters.